data/lua/core/map.lua

```diff
@@ -120,7 +162,7 @@ if wesnoth.kernel_type() == "Game Lua Kernel" then
 		elseif key == 2 then
 			return self.y
 		elseif type(key) ~= string or (#key > 0 and key[0] ~= '_') then
-			return hex_mt[key]
+			return hex_methods[key]
```

Comment:
There's an issue here, as a) I don't remember why the `~= string` test was added and b) the `~= string` test itself is wrong. I think it's out of scope for this PR though.